services/order.py

```diff
@@ -0,0 +1,40 @@
+from datetime import datetime
+from django.db import transaction
+from django.contrib.auth import get_user_model
+from django.db.models import QuerySet
+
+from db.models import Order, Ticket, MovieSession
+
+
+def create_order(
+        tickets: list[dict],
+        username: str,
+        date: str = None,
+) -> Order:
+    with transaction.atomic():
+        user = get_user_model().objects.get(username=username)
+        new_order = Order.objects.create(user=user)
+        if date:
+            created_at_format = datetime.strptime(date, "%Y-%m-%d %H:%M")
+            new_order.created_at = created_at_format
+            new_order.save()
+
+        for ticket_data in tickets:
+            row, seat, movie_session_id = ticket_data.values()
+            movie_session = MovieSession.objects.get(id=movie_session_id)
+            Ticket.objects.create(
+                order=new_order,
+                row=row,
+                seat=seat,
+                movie_session=movie_session,
+            )
+        return new_order
```

Comment:
You don't need to get movie_session here, in Ticket parameters just use movie_session_id=use movie_session_id and you will have same result.Tried by myself, it worked =)

Reply:
Thank you!)